cmd/dagger/dial_stdio.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dagger/dagger/engine"
+	"github.com/dagger/dagger/router"
+	"github.com/spf13/cobra"
+)
+
+var dialStdioCmd = &cobra.Command{
+	Use:    "dial-stdio",
+	Run:    DialStdio,
+	Hidden: true,
+}
+
+func DialStdio(cmd *cobra.Command, args []string) {
+	localDirs := getKVInput(localDirsInput)
+	startOpts := &engine.Config{
+		LocalDirs:  localDirs,
+		Workdir:    workdir,
+		ConfigPath: configPath,
+	}
+
+	err := engine.Start(context.Background(), startOpts, func(ctx context.Context, r *router.Router) error {
+		srv := http.Server{
+			Handler:           r,
+			ReadHeaderTimeout: 30 * time.Second,
+		}
+
+		closeCh := make(chan struct{})
+
+		l := &stdioConnListener{
+			closeCh: closeCh,
+		}
+
+		go func() {
+			err := srv.Serve(l)
+			if err != nil && err != http.ErrServerClosed {
+				fmt.Fprintf(os.Stderr, "http serve: error: %v\n", err)
+			}
+		}()
+
+		<-closeCh
+
+		return srv.Shutdown(ctx)
+	})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+var _ net.Listener = &stdioConnListener{}
+
+type stdioConnListener struct {
+	once    sync.Once
+	closeCh chan struct{}
+}
+
+func (l *stdioConnListener) Accept() (net.Conn, error) {
+	var conn net.Conn
+	l.once.Do(func() {
+		conn = &stdioConn{
+			closeCh: l.closeCh,
+		}
+	})
+	// Return stdio connection ONLY on the first `Accept()`
+	// since we can't multiplex on stdio.
```

Comment:
Are there any immediate needs for multiplexing? I think there may be in the future w/ services and/or filesync stuff, just wondering if I'm not thinking of already existing needs.